### DISCUSSION TITLE
Autocomplete for text-input box

### DISCUSSION THREAD

**Discussion by francesfeng** [widgetti/solara#499] [Ideas]
Hi team, 

Wonder if autocomplete is on the roadmap? features similar to this one https://mui.com/material-ui/react-autocomplete/

From the two scenarios described in Material UI, I think the 2nd item on the list is more meaningful. Because 1) can be achieved by using "Select" list. 

> The value for the textbox must be chosen from a predefined set of allowed values, e.g., a location field must contain a valid location name: [combo box](https://mui.com/material-ui/react-autocomplete/#combo-box).
The textbox may contain any arbitrary value, but it is advantageous to suggest possible values to the user, e.g., a search field may suggest similar or previous searches to save the user time: [free solo](https://mui.com/material-ui/react-autocomplete/#free-solo).

Essentially can extend the InputText element, like following? 

```
def InputText(
    label: str,
    value: Union[str, solara.Reactive[str]] = "",
    on_value: Callable[[str], None] = None,
    autocomplete: List[str] = None # a list of candidate values that dynamically generated as user typing in the input box
    ...
):
```

It is particularly useful for a recommendation system, like how it feels like when doing a google search.

**Comment by maartenbreddels** [comment]
Hi Francis,

someone on our discord used the vuetify autocomplete for this: https://v2.vuetifyjs.com/en/api/v-autocomplete

I hope this link works
https://discord.com/channels/1106593685241614489/1106593686223069309/1202928945226452993
